src/bin/pg_dump/pg_dump.c

```
@@ -1147,15 +1147,23 @@ expand_oid_patterns(SimpleStringList *patterns, SimpleOidList *oids)
 			if (*cp < '0' || *cp > '9')
 			{
 				if (cp != ch)
-					simple_oid_list_append(oids, atooid(strndup(ch, cp - ch)));
+				{
+					char tmp[11] = {0};
```

BTW, is it really intentional that you can use any non-digit character as the separator? Does this behave sanely if you e.g. have extra spaces in the string, like "1,  2"? Accepting any non-digit seems unnecessarily lenient anyway. If the user types something like "0x123" here, it's almost certainly bogus input and it'd be better to throw an error.

How about using strtok() to parse the list?